audb/core/load.py

```diff
@@ -746,6 +737,119 @@ def job(table):
     )
 
 
+def dependencies(
+        name: str,
+        *,
+        version: str = None,
+        tables: typing.Union[str, typing.Sequence[str]] = None,
+        media: typing.Union[str, typing.Sequence[str]] = None,
+        cache_root: str = None,
+        verbose: bool = False,
+) -> Dependencies:
+    r"""Database dependencies.
+
+    Args:
+        name: name of database
+        version: version of database
+        tables: include only tables matching the regular expression or
+            provided in the list
+        media: include only media matching the regular expression or
+            provided in the list
+        cache_root: cache folder where databases are stored.
+            If not set :meth:`audb.default_cache_root` is used
+        verbose: show debug messages
+
+    Returns:
+        dependency object
+
+    Raises:
+        ValueError: if table or media is requested
+            that is not part of the database
+
+    Example:
+        >>> deps = dependencies('emodb', version='1.2.0')
+        >>> deps.version('db.emotion.csv')
+        '1.1.0'
+
+    """
+    if version is None:
+        version = latest_version(name)
+
+    db_root = database_cache_root(
+        name,
+        version,
+        cache_root=cache_root,
+    )
+    deps_path = os.path.join(db_root, define.CACHED_DEPENDENCIES_FILE)
+
+    deps = Dependencies()
+
+    with FolderLock(db_root):
+        try:
+            deps.load(deps_path)
+        except (AttributeError, FileNotFoundError, ValueError, EOFError):
+            # If loading pickled cached file fails, load again from backend
+            backend = lookup_backend(name, version)
+            with tempfile.TemporaryDirectory() as tmp_root:
+                archive = backend.join(name, define.DB)
+                backend.get_archive(
+                    archive,
+                    tmp_root,
+                    version,
+                    verbose=verbose,
+                )
+                deps.load(os.path.join(tmp_root, define.DEPENDENCIES_FILE))
+                deps.save(deps_path)
+
+    if media is None and tables is None:
+        return deps
+
+    # Filter after requested tables and media
+    filtered_media = deps.media
+    filtered_tables = deps.table_ids
+
+    if media is not None:
+        filtered_media = _filter_media(
+            filtered_media,
+            media,
+            name,
+            version,
+        )
+
+    if tables is not None:
+        filtered_tables = _filter_tables(
+            filtered_tables,
+            tables,
+            name,
+            version,
+        )
+
+        table_media = set()
+        for table in filtered_tables:
+            df = load_table(
```

Comment:
I think this is problematic as it will download the table in database root folder and then later again in the flavor cache folder.

Reply:
So maybe we should not care about the intersection of table files and requested media at this stage and simply return all requested tables / media and do the additional filtering later?

Reply:
Yes, that was the reason why I had two separate approaches for `audb.info` and `audb.load()` before moving everything to `audb.dependencies()` as the first requires the tables outside of the flavor folder, the later inside the flavor folders.

But having separated code for the filtering was also not optimal. Would be nice if we could find a way to improve here.

Reply:
The problem is we need media files stored inside the requested tables, otherwise we cannot filter `deps._df` and this is needed as the current code relies on `deps` as the ground truth of requested files.

Reply:
Yes I got that. So you don't think it makes sense to find the intersection at a later stage but at least have a pre-filtering here?

Reply:
Yes and no, this was basically what I had implemented before as we need the filtering not only in `audb.load()`, but also inside all the `audb.info` functions.
So, if you like have another look at the implementation at commit https://github.com/audeering/audb/pull/210/commits/99bb927fe0d139be9ef5a11b72d4f5c5dd71c435

Reply:
Would it make sense we first merge your original solution and then try to optimize from there?

Reply:
Yes, good idea.

I will take another look and will start to create a few smaller pull request from here. Maybe by starting indeed adding the arguments to `audb.info` or by first starting optimizing already slightly what we have at the moment for filtering.

I will set this pull request to WIP.

Reply:
I added an alternative implementation in #215 that moved the functions as methods to `audb.Dependencies` and filter by downloaded tables in a last step with a different function as you proposed here